rdflib/plugins/stores/berkeleydb.py

```diff
@@ -483,6 +483,15 @@ def bind(self, prefix, namespace, override=True):
             self.__prefix[bound_namespace or namespace] = bound_prefix or prefix
             self.__namespace[bound_prefix or prefix] = bound_namespace or namespace
 
+    def unbind(self, prefix):
+        if prefix is None:
+            return
```

Comment:
As prefix should be supplied as a `str` I think it is best to omit this custom handling, if anything it can maybe be a `TypeError`, but I think avoiding custom handling for `prefix is None` is best. If someone does it they can live with the consequences if it does not fail.

```suggestion
```

Reply:
True, coherent type-hinting would resolve this more neatly and I was lazy. 

My excuse is - that from a meta-perspective I'm not sure that this PR is desirable in that adding an explicit `unbind` is arguably not worth the effort when the functionality already exists via `bind(<prefix>, None)` From this perspective, it's more of a documentation / docstring issue.

Reply:
> I'm not sure that this PR is desirable in that adding an explicit `unbind` is arguably not worth the effort when the functionality already exists via `bind(<prefix>, None)` From this perspective, it's more of a documentation / docstring issue.

If this is the case we can possibly implement this by checking if the underlying store implements `unbind`, and if not call `bind(, None)` - but I'm not sure if all stores will work with this either.

Reply:
> If this is the case we can possibly implement this by checking if the underlying store implements `unbind`, and if not call `bind(, None)` - but I'm not sure if all stores will work with this either.

Oh, looks like I was mistaken in believing that there was existing functionality for unbinding via binding to `None`

```py
def test_extant_graph_unbind(tmp_path: Path, store_name: str) -> None:
    graph = make_graph(tmp_path, store_name)
    graph.bind("", EGNSSUB_V0)
    check_ns(graph, {"": EGNSSUB_V0})

    graph.bind("", None)

    assert list(graph.namespaces()) == [
        ("", URIRef("http://example.com/sub/v0")),
        ("default1", URIRef("None")),
    ]


def test_extant_store_unbind(tmp_path: Path, store_name: str) -> None:
    graph = make_graph(tmp_path, store_name)
    graph.bind("", EGNSSUB_V0)
    check_ns(graph, {"": EGNSSUB_V0})

    graph.store.bind("", None)

    if store_name == "BerkeleyDB":
        assert list(graph.store.namespaces()) == [
            ("", URIRef("http://example.com/sub/v0"))
        ]
    else:
        assert list(graph.store.namespaces()) == [("", None)]
```

I guess that appreciably changes matters: the rebased PR *is* worth submitting, so coherent type-hinting it is then :smile:

Reply:
Okay, I've added more type hinting (perhaps clumsily, it's not one of my strong points). Had to protect berkeleydb from trying to apply `decode("utf-8)` to `None` although the tests do show that mypy will report the type error.

Reply:
> Okay, I've added more type hinting (perhaps clumsily, it's not one of my strong points).

Definitely not, sigh. Changed param `namespace` type to `str` throughout, reflecting contemporary usage, seems to satisfy everything except the `Type[DOAP]` in `test/util/earl.py`, so I've temporarily added type ignore to those. I guess `URIRef` and `Namespace` satisfy some behavioural criterion that allows them to be `str` whereas `Type` doesn't.